### PR TITLE
Added YouTube Metadata Reader

### PR DESCRIPTION
**Pull request by Troyusrex2**
This new package retrieves metadata from Youtube videos. It also allows combining that metadata with the transcript of the video. When using large numbers of videos for RAG AI, the transcript of the video was not enough. Having metadata, in particular the tags, but also statistics such as hit count, date and comment count were extremely useful for quick and effective use with LLMs. These can also easily be combined with an LLM generated summary of the transcript which I chose not to include here but will include when I create my example using this tool. 

It uses the youtube_transcript_api package and needs a google API key. 

It does not use the existing  llama-index-readers-youtube-transcript as using youtube_transcript_api directly was straightforward.

This is my first time updating any open source project. I tried hard to follow the guidelines, but if anything is not correct I am happy to fix it. 

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [X ] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [ X] No

## Type of Change

Please delete options that are not relevant.

- [ X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

I have used this as part of a larger project to collect and label thousands of Youtube videos from various creators. I had another programmer and a Copilot look it over as well. I wrote a separate testing app to call it remotely to check that it worked properly. 

- [X ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ X] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [ X] I have performed a self-review of my own code
- [ X] I have commented my code, particularly in hard-to-understand areas
- [ X] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ X] My changes generate no new warnings
- [ X] I have added tests that prove my fix is effective or that my feature works
- [ X] New and existing unit tests pass locally with my changes
- [ X] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/readers/llama-index-readers-youtube-metadata/pyproject.toml
Comment (nerdai) [inline llama-index-integrations/readers/llama-index-readers-youtube-metadata/pyproject.toml:1]: not sure how its missing but there should be `tool.llamahub` section here

File: llama-index-integrations/readers/llama-index-readers-youtube-metadata/tests/BUILD [hunk 1]
```diff
@@ -0,0 +1,5 @@
+python_sources()
+
+python_tests(
+    name="tests0",
+)
```
Comment (nerdai) [inline llama-index-integrations/readers/llama-index-readers-youtube-metadata/tests/BUILD:5]: did this get written by pants?

**Review by nerdai — COMMENTED** [review]
thanks for the contribution! left a couple comments to start

File: llama-index-integrations/readers/llama-index-readers-youtube-metadata/pyproject.toml
Comment (Troyusrex2) [reply llama-index-integrations/readers/llama-index-readers-youtube-metadata/pyproject.toml:1]: Thought I double checked that... but it's clearly not there. I will fix and re-commit.

File: llama-index-integrations/readers/llama-index-readers-youtube-metadata/tests/BUILD (hunk 1 above)
Comment (Troyusrex2) [reply llama-index-integrations/readers/llama-index-readers-youtube-metadata/tests/BUILD:5]: Yes. Written by Pants.

File: llama-index-integrations/readers/llama-index-readers-youtube-metadata/pyproject.toml
Comment (nerdai) [reply llama-index-integrations/readers/llama-index-readers-youtube-metadata/pyproject.toml:1]: thanks!

**Review by logan-markewich — APPROVED** [review]
